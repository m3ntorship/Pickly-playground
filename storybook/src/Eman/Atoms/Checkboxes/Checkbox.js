import React, { useState } from "react";
import PropTypes from "prop-types";

export const CHECKBOX = {
  SIZE: {
    DEFAULT: "Default",
    SMALL: "Small",
  },
};

export const Checkbox = ({ size, disabled }) => {
  const [isChecked, setIsChecked] = useState(true);

  const handleCheck = () => {
    setIsChecked(!isChecked);
  };

  const divSize = size === CHECKBOX.SIZE.DEFAULT ? "w-6 h-6" : "w-4 h-4";
  const iconSize = size === CHECKBOX.SIZE.DEFAULT ? "16" : "12";

  return (
    <>
      {isChecked && (
        <div
          className={`relative rounded-sm  flex justify-center  items-center bg-primary
          ${
            !disabled && "hover:bg-primary-shd8"
          } focus:outline-none focus:ring-2 focus:ring-offset-primary-shd6 ${divSize} 
          ${disabled && "bg-primary-shd7"}`}
          onClick={handleCheck}
        >
          <svg
            className="text-white"
            width={iconSize}
            height={iconSize}
            viewBox="0 0 16 16"
            fill="none"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.66632 10.3907L4.47099 8.19539L3.52832 9.13806L6.66632 12.2761L13.1377 5.80473L12.195 4.86206L6.66632 10.3907Z"
              fill="white"
            />
          </svg>
          <input
            onChange={(event) => setIsChecked(event.currentTarget.checked)}
            disabled={disabled}
            checked={isChecked}
            type="checkbox"
            className="opacity-0 absolute cursor-pointer w-full h-full"
          />
        </div>
      )}
      {!isChecked && (
        <div
          className={`relative rounded-sm flex justify-center items-center border border-grey-shd4 focus:outline-none  focus:ring-2 focus:ring-offset-grey-shd2 ${
            !disabled && "hover:border  hover:border-grey-shd1"
          } ${divSize} ${disabled && "border-grey-shd6"}`}
          onClick={handleCheck}
        >
          <input
            onChange={(event) => setIsChecked(event.currentTarget.checked)}
            disabled={disabled}
            checked={isChecked}
            type="checkbox"
            className="opacity-0 absolute cursor-pointer w-full h-full"
          />
        </div>
      )}
    </>
  );
};

Checkbox.propTypes = {
  disabled: PropTypes.bool,
  size: PropTypes.oneOf(["Default", "Small"]),
};
