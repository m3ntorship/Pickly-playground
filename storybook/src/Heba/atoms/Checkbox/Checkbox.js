import React from "react";
import PropTypes from "prop-types";

export const CHECKBOX = {
  SIZE: {
    DEFAULT: "Default",
    SMALL: "Small",
  },
};

export const Checkbox = ({ status, size }) => {
  return (
    <>
      {size === CHECKBOX.SIZE.DEFAULT && status && (
        <button className="w-6 h-6 rounded-sm bg-primary flex justify-center  items-center">
          <svg
            className="text-white"
            width="16"
            height="16"
            viewBox="0 0 16 16"
            fill="none"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.66632 10.3907L4.47099 8.19539L3.52832 9.13806L6.66632 12.2761L13.1377 5.80473L12.195 4.86206L6.66632 10.3907Z"
              fill="white"
            />
          </svg>
        </button>
      )}
      {size === CHECKBOX.SIZE.DEFAULT && !status && (
        <button className="w-6 h-6 border border-grey-shd4 rounded-sm flex justify-center items-center "></button>
      )}
      {size === CHECKBOX.SIZE.SMALL && status && (
        <button className="w-4 h-4 rounded-sm bg-primary flex justify-center  items-center">
          <svg
            width="12"
            height="12"
            viewBox="0 0 12 12"
            fill="none"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4.99998 7.79298L3.35348 6.14648L2.64648 6.85348L4.99998 9.20698L9.85348 4.35348L9.14648 3.64648L4.99998 7.79298Z"
              fill="white"
            />
          </svg>
        </button>
      )}
      {size === CHECKBOX.SIZE.SMALL && !status && (
        <button className="w-4 h-4 border border-grey-shd4 rounded-sm flex justify-center items-center "></button>
      )}
    </>
  );
};

Checkbox.propTypes = {
  status: PropTypes.bool,
  size: PropTypes.oneOf(["Default", "Small"]),
};
