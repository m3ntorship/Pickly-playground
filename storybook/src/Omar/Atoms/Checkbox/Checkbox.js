import React, { useState } from "react";
import PropTypes from "prop-types";
const Checkbox = ({ disabled }) => {
  const [isChecked, setIsChecked] = useState(true);
  return (
    <>
      {isChecked && (
        <label className="relative" data-testid="label">
          <input
            className={`w-l h-l appearance-none outline-none border border-grey-shd4 rounded-sm text-white align-baseline cursor-pointer bg-primary ${
              disabled && "bg-primary-shd7"
            } ${!disabled && "hover:bg-primary-shd8"} `}
            type="checkbox"
            data-testid="input"
            onChange={(event) => setIsChecked(event.currentTarget.checked)}
            disabled={disabled}
            checked={isChecked}
          />
          <svg
            style={{ left: "7px" }}
            className={`absolute bottom-3`}
            width="11"
            height="9"
            viewBox="0 0 11 9"
            fill="none"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3.66632 6.39073L1.47099 4.19539L0.52832 5.13806L3.66632 8.27606L10.1377 1.80473L9.19499 0.862061L3.66632 6.39073Z"
              fill="white"
            />
          </svg>
        </label>
      )}
      {!isChecked && (
        <label className="relative" data-testid="label">
          <input
            className={`w-l h-l appearance-none outline-none border border-grey-shd4 rounded-sm text-white align-baseline cursor-pointer ${
              !disabled && "hover:border  hover:border-grey-shd1"
            }`}
            type="checkbox"
            data-testid="input"
            onChange={(event) => setIsChecked(event.currentTarget.checked)}
            disabled={disabled}
            checked={isChecked}
          />
        </label>
      )}
    </>
  );
};
Checkbox.propTypes = {
  disabled: PropTypes.bool,
};

export default Checkbox;
