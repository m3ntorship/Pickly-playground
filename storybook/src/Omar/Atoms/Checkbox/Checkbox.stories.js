import React from "react";
import { Checkbox, CHECKBOX } from "./Checkbox";

export default {
  title: "Omar/Atoms/Checkbox",
  component: Checkbox,
  argTypes: {
    disabled: {
      control: {
        type: "boolean",
      },
    },
  },
};

const Template = (args) => <Checkbox {...args} />;
export const checkbox = Template.bind({});
checkbox.args = {
  disabled: false,
  size: CHECKBOX.SIZE.LARGE,
};
