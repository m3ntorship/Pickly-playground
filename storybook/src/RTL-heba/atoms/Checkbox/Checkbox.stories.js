import React from "react";
import { Checkbox, CHECKBOX } from "../Checkbox/Checkbox";

export default {
  title: "Heba/atoms/Checkboxes",
  component: Checkbox,
  argTypes: {
    status: {
      control: {
        type: "boolean",
      },
    },
    size: {
      control: {
        type: "select",
        options: [CHECKBOX.SIZE.DEFAULT, CHECKBOX.SIZE.SMALL],
      },
    },
  },
};

const Template = (args) => <Checkbox {...args} />;

export const checkbox = Template.bind({});
checkbox.args = {
  status,
  size: CHECKBOX.SIZE.DEFAULT,
};
