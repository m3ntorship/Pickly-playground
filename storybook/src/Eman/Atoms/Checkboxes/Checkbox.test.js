import React from "react";
import renderer from "react-test-renderer";
import {Checkbox} from "./Checkbox";

describe("Testing Checkbox", () => {
  it("Testing Checkbox size Default", () => {
    const tree = renderer.create(<Checkbox size="Default" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Testing Checkbox size Small", () => {
    const tree = renderer.create(<Checkbox size="Small" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Testing Checkbox with no props", () => {
    const tree = renderer.create(<Checkbox />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
