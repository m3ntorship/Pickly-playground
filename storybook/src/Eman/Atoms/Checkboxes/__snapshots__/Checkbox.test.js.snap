// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Checkbox Testing Checkbox size Default 1`] = `
<div
  className="relative rounded-sm  flex justify-center  items-center bg-primary
          hover:bg-primary-shd8 focus:outline-none focus:ring-2 focus:ring-offset-primary-shd6 w-6 h-6 
          undefined"
  onClick={[Function]}
>
  <svg
    className="text-white"
    fill="none"
    height="16"
    viewBox="0 0 16 16"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M6.66632 10.3907L4.47099 8.19539L3.52832 9.13806L6.66632 12.2761L13.1377 5.80473L12.195 4.86206L6.66632 10.3907Z"
      fill="white"
    />
  </svg>
  <input
    checked={true}
    className="opacity-0 absolute cursor-pointer w-full h-full"
    onChange={[Function]}
    type="checkbox"
  />
</div>
`;

exports[`Testing Checkbox Testing Checkbox size Small 1`] = `
<div
  className="relative rounded-sm  flex justify-center  items-center bg-primary
          hover:bg-primary-shd8 focus:outline-none focus:ring-2 focus:ring-offset-primary-shd6 w-4 h-4 
          undefined"
  onClick={[Function]}
>
  <svg
    className="text-white"
    fill="none"
    height="12"
    viewBox="0 0 16 16"
    width="12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M6.66632 10.3907L4.47099 8.19539L3.52832 9.13806L6.66632 12.2761L13.1377 5.80473L12.195 4.86206L6.66632 10.3907Z"
      fill="white"
    />
  </svg>
  <input
    checked={true}
    className="opacity-0 absolute cursor-pointer w-full h-full"
    onChange={[Function]}
    type="checkbox"
  />
</div>
`;

exports[`Testing Checkbox Testing Checkbox with no props 1`] = `
<div
  className="relative rounded-sm  flex justify-center  items-center bg-primary
          hover:bg-primary-shd8 focus:outline-none focus:ring-2 focus:ring-offset-primary-shd6 w-4 h-4 
          undefined"
  onClick={[Function]}
>
  <svg
    className="text-white"
    fill="none"
    height="12"
    viewBox="0 0 16 16"
    width="12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M6.66632 10.3907L4.47099 8.19539L3.52832 9.13806L6.66632 12.2761L13.1377 5.80473L12.195 4.86206L6.66632 10.3907Z"
      fill="white"
    />
  </svg>
  <input
    checked={true}
    className="opacity-0 absolute cursor-pointer w-full h-full"
    onChange={[Function]}
    type="checkbox"
  />
</div>
`;
